charts/velero/templates/_helpers.tpl

```diff
@@ -89,7 +89,7 @@ Create the backup storage location provider
 */}}
 {{- define "velero.backupStorageLocation.provider" -}}
 {{- with .Values.configuration -}}
-{{ default .backupStorageLocation.provider .provider }}
+{{ default .provider .backupStorageLocation.provider }}
```

Comment:
should this be `{{ default .provider .backupStorageLocation.name }}`? `.backupStorageLocation.name` is what I see documented in `values.yaml` and the docs. However, I also see on L90 that we're using `{{- define "velero.backupStorageLocation.provider" -}}`, and in the backupstoragelocation template we reference `backupStorageLocation.provider` (https://github.com/vmware-tanzu/helm-charts/blob/master/charts/velero/templates/backupstoragelocation.yaml#L12).

Not entirely sure but it seems like there's an inconsistency somewhere.

Reply:
hey @skriss I see name and provider being a bit decoupled - name is used for the name of a resource while the provider is used by the plugin for config.

To use name you actually introduce complexity - for example when using digital ocean you would set your provider as `digitalocean.com/velero` which is not a valid kubernetes resource name because of the `/`

Reply:
I totally agree they *should* be separate things. I'm just trying to ensure that (a) we're not making a backwards-incompatible change if we were previously using the "name" value to fill in the provider and now we're no longer doing so; and (b) that the docs are accurate.

I can't really tell at a glance if we're making a backwards-incompatible change here, but it seems like probably not?

If not, then let's please just document that there is now a `backupStorageLocation.provider` that is separate from `backupStorageLocation.name`.